src/js/components/Tag/Tag.js

```diff
@@ -23,27 +35,61 @@ const Tag = forwardRef(
       round: theme.tag.size?.[size]?.round || theme.tag.round,
       ...rest,
     };
-    const contents = (
+    const displayName =
+      truncate && name.length > truncateNumChars
+        ? `${name.substring(0, truncateNumChars - 3)}...`
+        : name;
+    const displayValue =
+      truncate && value.length > truncateNumChars
+        ? `${value.substring(0, truncateNumChars - 3)}...`
+        : value;
```

Comment:
FYI I'd rather see this done via some element width constraint and the css ellipsis truncation rather than string truncation based on number of chars.

Reply:
I agree with @MikeKingdom's feedback. You should be using CSS-only truncation, not `substring`. In general, if it is possible to accomplish something using CSS-only solutions, you should try to make it work that way.

Reply:
Hey @MikeKingdom and @samuelgoff thanks for the feedback. I have updated the pr to use width based constraints and let the the Text elements handle the truncation internally. Also, tips are "stacked" vertically so they don't get crazy wide and I've added window.resize update for responsive truncation. If you feel like this is ready to go, I'll take it out of draft status, or if you have further feedback, I would be happy to address any other issues.